compiler/src/dotty/tools/dotc/ast/MainProxies.scala

```diff
@@ -94,8 +94,20 @@ object MainProxies {
       val body = Try(call, handler :: Nil, EmptyTree)
       val mainArg = ValDef(nme.args, TypeTree(defn.ArrayType.appliedTo(defn.StringType)), EmptyTree)
         .withFlags(Param)
+      /** Replace typed `Ident`s that have been typed with a TypeSplice with the reference to the symbol.
+       *  The annotations will be retype-checked in another scope that may not have the same imports.
+       */
+      def insertTypeSplices = new TreeMap {
+          override def transform(tree: Tree)(using Context): Tree = tree match
```

Comment:
It's weird the `@unchecked` is needed below, maybe it wouldn't be with:
```suggestion
          override def transform(tree: tpd.Tree)(using Context): tpd.Tree = tree match
```

Reply:
To make that work I would also need to make it a `new tpd.TreeMap`.

The problem with that is that `TypedSplice(tree)` is not a `tpd.Tree`. I would need to cast it.

Reply:
oh I see, because we're actually going from tpd.Tree to untpd.Tree, and neither tpd.TreeMap nor untpd.TreeMap have the correct signature, I guess the current solution is fine then?

Reply:
I find the current version cleaner

Reply:
This logic that converts typed annotation trees in partially untyped trees with `TypedSplice `s only for `Ident` does not work in all cases.

I think it can be simplified to just use a `TypedSplice` for the whole annotation tree, given that it gets the correct owner.